src/core/xccl_init.c

```diff
@@ -35,20 +56,20 @@ static int
 callback(struct dl_phdr_info *info, size_t size, void *data)
 {
     char *str;
-    xccl_lib_t *lib = (xccl_lib_t*)data;
     if (NULL != (str = strstr(info->dlpi_name, "libxccl.so"))) {
         int pos = (int)(str - info->dlpi_name);
-        lib->lib_path = (char*)malloc(pos+8);
-        strncpy(lib->lib_path, info->dlpi_name, pos);
-        lib->lib_path[pos] = '\0';
-        strcat(lib->lib_path, "xccl");
+        free(xccl_lib_global_config.team_lib_path);
```

Comment:
plz make sure this is a valid "free" on the first call of the "callback". It should contain "" empty string on that first call if i understand correctly - was it malloced by the ucs config parser?

Reply:
ucs config assigns default value using strdup which in turn allocates memory, so this free is valid